### ISSUE TITLE
max_screen_size param do not work in the code

### ISSUE THREAD

**Issue by nickle-fang** [graphdeco-inria/gaussian-splatting#544]
I found that `max_screen_size` param do not work in the code. I wonder whether it is a feature or bug here.

- In `densify_and_clone` and `densify_and_split` the `self.max_radii2D` will all be set to 0, so the max_screen_size below filter will not work at all.
- 0 setting code is at [here](https://github.com/graphdeco-inria/gaussian-splatting/blob/2eee0e26d2d5fd00ec462df47752223952f6bf4e/scene/gaussian_model.py#L347C18-L347C18)

```
def densify_and_prune(self, max_grad, min_opacity, extent, max_screen_size):
    grads = self.xyz_gradient_accum / self.denom
    grads[grads.isnan()] = 0.0
 
    self.densify_and_clone(grads, max_grad, extent)
    self.densify_and_split(grads, max_grad, extent)

    prune_mask = (self.get_opacity < min_opacity).squeeze()
    if max_screen_size:
        big_points_vs = self.max_radii2D > max_screen_size
        big_points_ws = self.get_scaling.max(dim=1).values > 0.1 * extent
        prune_mask = torch.logical_or(torch.logical_or(prune_mask, big_points_vs), big_points_ws)
    self.prune_points(prune_mask)
```

**Comment by ducksoup** [comment]
Can confirm that this is something that also came up for me as I was going through the code cross-checking details from the paper. Notably, the results shown in the paper cannot be reproduced if one fixes this bug, as the point cloud is pruned too aggressively with the given hyper-parameters.

P.S.: Another one detail that is different w.r.t. the paper is that the warm-up procedure described in Sec. 7.1 is, as far as I can tell, not really implemented in the code.

**Comment by qxforever** [comment]
In #123, "It is a bug that was also present in evaluation".
